### ISSUE TITLE
Video Dimension & FPS setting do not work (P2P or Group)

### ISSUE THREAD

**Issue by anooshcnayak** [twilio/video-quickstart-android#584]
I have checked all the Best practices & the article on Developing High-Quality Video Applications.

### Description
Altering VideoDimension & FPS do not have any significance. **P2P** Room - **2** Participants & also **Group** - 2 Participants

### Settings:
**1 FPS** & 5 x 5 Video Dimension. MaxAudioBitrate - 16, MaxVideoBitrate - 0 (as suggested in the best practices for p2p)

#### Code

```java
VideoDimensions dimensions = new VideoDimensions(5, 5);
videoConstraints = VideoConstraints.Builder()
                .minVideoDimensions(dimensions)
                .maxVideoDimensions(dimensions)
                .minFps(1)
                .maxFps(1)
                .build()
localVideoTrack = LocalVideoTrack.create(getContext(), enableVideo, cameraCapturer, videoConstraints);
```

#### Expected Behavior

Data consumption to be minimal, Resolution, FPS changes to be reflected. getStats API to provide the given config

#### Actual Behavior

It sends completely different FPS & resolution irrespective of what I set in the configuration. I have no control over the resolution & fps

Inferred further with the getStats Api of the Room object - Output below:
```json
{"F4dCeCfEcCe7bF7f6fd436B280110C3D":{"localVideoTrackStats":[{"roundTripTime":27,"packetsSent":4772,"bytesSent":4914497,"trackSid":"MT471c935b84e37a28af516d7d5222560d","timestamp":1606133744287.274,"ssrc":"1831751530","packetsLost":0,"codec":"VP8","captureDimensions":{"height":640,"width":480},"frameRate":30,"dimensions":{"width":480,"height":640}}],"localAudioTrackStats":[{"roundTripTime":27,"packetsSent":2529,"bytesSent":237386,"trackSid":"MT78bcdf73f53192d1a5b4b4d5a0ac2b35","timestamp":1606133744287.274,"ssrc":"4161008162","packetsLost":0,"codec":"opus","jitter":1,"audioLevel":32767}],"remoteVideoTrackStats":[{"packetsReceived":10270,"bytesReceived":11165321,"trackSid":"MT72f290acf4049fb66dc0f3eba5f2a04c","timestamp":1606133744287.274,"ssrc":"3354574052","packetsLost":0,"codec":"VP8","frameRate":30,"dimensions":{"width":480,"height":640}}],"remoteAudioTrackStats":[{"packetsReceived":2580,"bytesReceived":225387,"trackSid":"MT17a303b148cf71a0485c22cbf4fed44f","timestamp":1606133744287.274,"ssrc":"1745501521","packetsLost":0,"codec":"opus","jitter":3,"audioLevel":1022}]}}
```

**Note:** `"captureDimensions":{"height":640,"width":480},"frameRate":30,"dimensions":{"width":480,"height":640}`

#### Reproduces how Often

Everytime

### Versions

Latest Version.

#### Video Android SDK

Android 9

#### Android API

29

#### Android Device

Oneplus 6T

**Comment by aaalaniz** [comment]
Hey @anooshcnayak 

The `VideoConstraints` can be a little misleading with respect to the expected behavior. The 5.0 SDK determines the capture format based on the return value of `VideoCapturer#getSupportedFormats()` and the provided `VideoConstraints`. If the SDK can not find a format within the specified constraints, then the default format (640x480@30FPS) will be used.

In the example you provided, the SDK behaves as expected. None of our provided `VideoCapturer`s can capture at 5x5@1FPS so the SDK will fall back to VGA as you have observed. If you are trying to reduce the capture size in the 5.0 SDK, then I suggest the following approach.

```
VideoDimensions dimensions = new VideoDimensions(320, 240);
videoConstraints = VideoConstraints.Builder()
                .maxVideoDimensions(dimensions)
                .maxFps(15)
                .build()
localVideoTrack = LocalVideoTrack.create(getContext(), enableVideo, cameraCapturer, videoConstraints);
```

While this snippet does not provide the same behavior as the original one you provided, it does ensure that your video track will capture at a lower resolution and framerate as most cameras should be able to satisfy these constraints.

---

Please note that `VideoConstraints` have been removed in the [Video Android 6.0 SDK](https://www.twilio.com/docs/video/600-preview) in favor of specifying a `VideoFormat`. You can now specify a capture format when creating a local video track and the VideoCapturer is responsible for capturing at a format as close as possible to the specified format. So in the example you provided, you could perform the following:

```
VideoFormat videoFormat = new VideoFormat(new VideoDimensions(5, 5), 1);
localVideoTrack = LocalVideoTrack.create(getContext(), enableVideo, cameraCapturer, videoFormat);
```

In 6.0, the SDK will pick a format as close as possible to 5x5@1FPS. The actual format will vary by device as each camera's supported formats differ.

I highly recommend migrating your application to 6.0 as I think you will find the updated capturing and rendering facilities much easier to customize to your liking. For more details about how to migrate your application, [reference our migration guide](https://www.twilio.com/docs/video/migrating-5x-6x).

I will close this issue but feel free to reopen if you have any follow-up questions or comments.

Thanks!

**Comment by anooshcnayak** [comment]
Hey @aaalaniz, Thanks for the help. I have migrated to 6.0 - Looks fine. I have set 5x5 @ 1 fps & I get 144x176 @15 fps (on the same device). Does this mean that I **cannot** scale down further the resolution or the fps( say 50 * 50) ?

Also, max video bitrate & maxAudioBitrate do not seem to work as expected. I have set both to 1, 1 Kilobits - the minimal value possible but I still see 15 Kilo**bytes** per second data getting pushed out. Should I raise another issue here for this? Am I missing anything?

**Comment by aaalaniz** [comment]
Hey @anooshcnayak 

>I have set 5x5 @ 1 fps & I get 144x176 @15 fps (on the same device). Does this mean that I cannot scale down further the resolution or the fps( say 50 * 50) ?

Scaling down further does require some additional work. Reference the following class as an example of how to scale down the existing CameraCapturer. This same approach could be used with any of the SDK VideoCapturers.  

```kotlin
class CroppedCameraCapturer(context: Context, cameraId: String) :
    CameraCapturer(context, cameraId) {

    class Cropper(private val capturerObserver: CapturerObserver): CapturerObserver {
        override fun onCapturerStarted(success: Boolean) {
            capturerObserver.onCapturerStarted(success)
        }

        override fun onCapturerStopped() {
            capturerObserver.onCapturerStopped()
        }

        override fun onFrameCaptured(frame: VideoFrame?) {
            frame?.let {
                // You can configure these values to fit your use case
                val cropWidth = 50
                val cropHeight = 50
                val scaledWidth = 50
                val scaledHeight = 50
                val cropX = (it.buffer.width / 2) - (cropWidth / 2)
                val cropY = (it.buffer.height / 2) - (cropHeight / 2)
                val croppedBuffer = it.buffer.cropAndScale(cropX, cropY, cropWidth, cropHeight, scaledWidth, scaledHeight)
                capturerObserver.onFrameCaptured(VideoFrame(croppedBuffer, frame.rotation, frame.timestampNs))
            }
        }
    }

    override fun initialize(
        surfaceTextureHelper: SurfaceTextureHelper,
        context: Context,
        capturerObserver: CapturerObserver
    ) {
        super.initialize(surfaceTextureHelper, context, Cropper(capturerObserver))
    }
}
```

In terms of adjusting the frame rate, you are restricted by the supported frame rate from the device's camera. However, you are welcome to just drop frames yourself in an effort to reduce the number of frames sent over the wire. This is as simple as not reporting to the `capturerObserver` when a frame has been captured. The frame will still be captured from the camera, but it will not be reported to any video sinks.

**Comment by anooshcnayak** [comment]
Hey @aaalaniz  There seems to some issue with this.

Note: I have extended `Camera2Capturer` class of com.twilio.video
The Camera Crashes & I get this error:

```
2020-12-09 17:36:08.465 21069-24876/com.twilioVideoProto E/tvi.webrtc.Logging: CameraStatistics: Camera freezed.
2020-12-09 17:36:08.470 21069-24876/com.twilioVideoProto E/Camera2Capturer: Camera failure. Client must return video buffers.
    com.twilio.video.Camera2Capturer$Exception: Camera failure. Client must return video buffers.
        at com.twilio.video.Camera2Capturer$1.onCameraFreezed(Camera2Capturer.java:85)
        at tvi.webrtc.CameraVideoCapturer$CameraStatistics$1.run(CameraVideoCapturer.java:131)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.os.HandlerThread.run(HandlerThread.java:67)
```

The problems seems to be with `val croppedBuffer = it.buffer.cropAndScale(cropX, cropY, cropWidth, cropHeight, scaledWidth, scaledHeight)` - It works without this line, i.e, creating new instance of videoframe with same buffer,rotation,timestamp & passing it to onFrameCaptured of observer

The original Video Frame obtained in onFrameCaputred has the resolution 176x144.
After mutation, `cropX 63 cropY47 CroppedBuffer Width & Height 50 50`
Looks, like it's cropping it correctly, but somehow the camera crashes later.

I'm trying to further debug. Could you help me here?

Complete Logs:

```
2020-12-09 17:43:50.616 21595-21595/com.twilioVideoProto I/tvi.webrtc.Logging: EglBase14Impl: SDK version: 29. isEGL14Supported: true
2020-12-09 17:43:50.617 21595-21595/com.twilioVideoProto I/tvi.webrtc.Logging: EglBase14Impl: Using OpenGL ES version 2
2020-12-09 17:43:50.632 21595-21595/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManagerExternal: Sample rate is set to 48000 Hz
2020-12-09 17:43:50.632 21595-21595/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManagerExternal: Sample rate is set to 48000 Hz
2020-12-09 17:43:50.632 21595-21595/com.twilioVideoProto I/tvi.webrtc.Logging: JavaAudioDeviceModule: createAudioDeviceModule
2020-12-09 17:43:50.632 21595-21595/com.twilioVideoProto I/tvi.webrtc.Logging: JavaAudioDeviceModule: HW NS will be used.
2020-12-09 17:43:50.632 21595-21595/com.twilioVideoProto I/tvi.webrtc.Logging: JavaAudioDeviceModule: HW AEC will be used.
2020-12-09 17:43:50.634 21595-21595/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioEffectsExternal: ctor@[name=main, id=2]
2020-12-09 17:43:50.634 21595-21595/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioRecordExternal: ctor@[name=main, id=2]
2020-12-09 17:43:50.635 21595-21595/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioTrackExternal: ctor@[name=main, id=2]
2020-12-09 17:43:50.648 21595-21595/com.twilioVideoProto W/VideoCapabilities: Unsupported mime video/divx
2020-12-09 17:43:50.649 21595-21595/com.twilioVideoProto W/VideoCapabilities: Unsupported mime video/divx311
2020-12-09 17:43:50.649 21595-21595/com.twilioVideoProto W/VideoCapabilities: Unsupported mime video/divx4
2020-12-09 17:43:50.652 21595-21595/com.twilioVideoProto W/VideoCapabilities: Unrecognized profile/level 0/3 for video/mpeg2
2020-12-09 17:43:50.653 21595-21595/com.twilioVideoProto W/VideoCapabilities: Unrecognized profile/level 0/3 for video/mpeg2
2020-12-09 17:43:50.655 21595-21595/com.twilioVideoProto W/VideoCapabilities: Unsupported mime video/x-ms-wmv
2020-12-09 17:43:50.655 21595-21595/com.twilioVideoProto W/VideoCapabilities: Unsupported mime video/x-ms-wmv
2020-12-09 17:43:50.693 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager: ctor@[name=Thread-5, id=12384]
2020-12-09 17:43:50.694 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager: Sample rate is set to 48000 Hz
2020-12-09 17:43:50.699 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioEffects: canUseAcousticEchoCanceler: true
2020-12-09 17:43:50.699 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioEffects: canUseNoiseSuppressor: true
2020-12-09 17:43:50.700 21595-26034/com.twilioVideoProto W/tvi.webrtc.Logging: WebRtcAudioManager: AAudio support is currently disabled on all devices!
2020-12-09 17:43:50.701 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager: Android SDK: 29, Release: 10, Brand: OnePlus, Device: OnePlus5T, Id: QKQ1.191014.012, Hardware: qcom, Manufacturer: OnePlus, Model: ONEPLUS A5010, Product: OnePlus5T
2020-12-09 17:43:50.703 21595-26034/com.twilioVideoProto I/AudioManager: In isSpeakerphoneOn(), calling application: com.twilioVideoProto
2020-12-09 17:43:50.704 21595-26034/com.twilioVideoProto I/AudioManager: In isBluetoothScoOn(), calling application: com.twilioVideoProto
2020-12-09 17:43:50.704 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager: Audio State: audio mode: MODE_NORMAL, has mic: true, mic muted: false, music active: false, speakerphone: false, BT SCO: false
2020-12-09 17:43:50.704 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager: Audio State: 
2020-12-09 17:43:50.704 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager:   fixed volume=false
2020-12-09 17:43:50.705 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager:   STREAM_VOICE_CALL: volume=6, max=6, muted=false
2020-12-09 17:43:50.705 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager:   STREAM_MUSIC: volume=15, max=30, muted=false
2020-12-09 17:43:50.706 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager:   STREAM_RING: volume=1, max=7, muted=false
2020-12-09 17:43:50.707 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager:   STREAM_ALARM: volume=0, max=7, muted=true
2020-12-09 17:43:50.707 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager:   STREAM_NOTIFICATION: volume=1, max=7, muted=false
2020-12-09 17:43:50.708 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager:   STREAM_SYSTEM: volume=1, max=7, muted=false
2020-12-09 17:43:50.710 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager: Audio Devices: 
2020-12-09 17:43:50.710 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager:   TYPE_BUILTIN_EARPIECE(out): channels=[1], encodings=[2], sample rates=[48000], id=2
2020-12-09 17:43:50.710 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager:   TYPE_BUILTIN_SPEAKER(out): channels=[2], encodings=[2], sample rates=[48000], id=3
2020-12-09 17:43:50.710 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager:   TYPE_TELEPHONY(out): channels=[1, 2], encodings=[2], sample rates=[8000, 16000], id=8
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager:   TYPE_BUILTIN_MIC(in): channels=[1, 2], encodings=[2], sample rates=[8000, 11025, 12000, 16000, 22050, 24000, 32000, 44100, 48000], id=14
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager:   TYPE_TELEPHONY(in): channels=[1, 2], encodings=[2], sample rates=[8000, 11025, 12000, 16000, 22050, 24000, 32000, 44100, 48000], id=11
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager:   TYPE_BUILTIN_MIC(in): channels=[1, 2], encodings=[2], sample rates=[8000, 11025, 12000, 16000, 22050, 24000, 32000, 44100, 48000], id=15
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager:   TYPE_FM_TUNER(in): channels=[1, 2], encodings=[2], sample rates=[8000, 11025, 12000, 16000, 22050, 24000, 32000, 44100, 48000], id=12
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto D/OpenSLESPlayer: ctor[tid=26034]
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: ctor[tid=26034]
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto D/OpenSLESPlayer: AttachAudioBuffer
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto D/OpenSLESPlayer: SetPlayoutSampleRate(48000)
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto D/OpenSLESPlayer: SetPlayoutChannels(1)
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto D/OpenSLESPlayer: AllocateDataBuffers
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto D/OpenSLESPlayer: native buffer size: 192
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto D/OpenSLESPlayer: native buffer size in ms: 4.00
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: AttachAudioBuffer
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: SetRecordingSampleRate(48000)
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: SetRecordingChannels(1)
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: AllocateDataBuffers
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: frames per native buffer: 192
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: frames per 10ms buffer: 480
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: bytes per native buffer: 384
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: native sample rate: 48000
2020-12-09 17:43:50.711 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager: init@[name=Thread-5, id=12384]
2020-12-09 17:43:50.712 21595-26034/com.twilioVideoProto I/tvi.webrtc.Logging: WebRtcAudioManager: audio mode is: MODE_NORMAL
2020-12-09 17:43:50.712 21595-26034/com.twilioVideoProto D/OpenSLESPlayer: Init[tid=26034]
2020-12-09 17:43:50.712 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: Init[tid=26034]
2020-12-09 17:43:50.712 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: EnableBuiltInAEC(1)
2020-12-09 17:43:50.712 21595-26034/com.twilioVideoProto E/OpenSLESRecorder: Not implemented
2020-12-09 17:43:50.713 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: EnableBuiltInNS(1)
2020-12-09 17:43:50.713 21595-26034/com.twilioVideoProto E/OpenSLESRecorder: Not implemented
2020-12-09 17:43:50.758 21595-21595/com.twilioVideoProto I/tvi.webrtc.Logging: EglRenderer: Initializing EglRenderer
2020-12-09 17:43:50.759 21595-26043/com.twilioVideoProto I/tvi.webrtc.Logging: EglRenderer: EglBase.create shared context
2020-12-09 17:43:50.760 21595-26043/com.twilioVideoProto I/tvi.webrtc.Logging: EglBase14Impl: Using OpenGL ES version 2
2020-12-09 17:43:50.761 21595-21595/com.twilioVideoProto I/tvi.webrtc.Logging: EglRenderer: setMirrorHorizontally: false
2020-12-09 17:43:50.786 21595-21595/com.twilioVideoProto I/tvi.webrtc.Logging: EglRenderer: setLayoutAspectRatio: 0.5304519
2020-12-09 17:43:50.792 21595-26045/com.twilioVideoProto D/SoLoader: About to load: libimagepipeline.so
2020-12-09 17:43:50.792 21595-26045/com.twilioVideoProto D/SoLoader: libimagepipeline.so not found on /data/data/com.twilioVideoProto/lib-main
2020-12-09 17:43:50.792 21595-26045/com.twilioVideoProto D/SoLoader: libimagepipeline.so found on /data/app/com.twilioVideoProto-GcrsbOlUW14ywnhQPp_uXQ==/lib/arm64
2020-12-09 17:43:50.792 21595-26045/com.twilioVideoProto D/SoLoader: Not resolving dependencies for libimagepipeline.so
2020-12-09 17:43:50.794 21595-26045/com.twilioVideoProto D/SoLoader: Loaded: libimagepipeline.so
2020-12-09 17:43:50.804 21595-21595/com.twilioVideoProto I/tvi.webrtc.Logging: SurfaceEglRenderer: : surfaceChanged: format: 4 size: 270x509
2020-12-09 17:43:50.820 21595-21595/com.twilioVideoProto I/CameraManagerGlobal: Connecting to camera service
2020-12-09 17:43:50.821 21595-21595/com.twilioVideoProto W/CameraManagerGlobal: [soar.cts] ignore the status update of camera: 2
2020-12-09 17:43:50.821 21595-21595/com.twilioVideoProto W/CameraManagerGlobal: [soar.cts] ignore the status update of camera: 3
2020-12-09 17:43:50.822 21595-25795/com.twilioVideoProto W/CameraManagerGlobal: ignore the torch status update of camera: 2
2020-12-09 17:43:50.822 21595-25795/com.twilioVideoProto W/CameraManagerGlobal: ignore the torch status update of camera: 3
2020-12-09 17:43:50.855 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: EglBase14Impl: Using OpenGL ES version 2
2020-12-09 17:43:50.862 21595-21595/com.twilioVideoProto I/tvi.webrtc.Logging: CameraCapturer: startCapture: 50x50@30
2020-12-09 17:43:50.864 21595-21595/com.twilioVideoProto I/tvi.webrtc.Logging: EglRenderer: setMirrorHorizontally: true
2020-12-09 17:43:50.865 21595-21595/com.twilioVideoProto D/BluetoothHeadset: Binding service...
2020-12-09 17:43:50.865 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: Camera2Session: Create new camera2 session on camera 1
2020-12-09 17:43:50.866 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: Camera2Session: start
2020-12-09 17:43:50.876 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: Camera2Session: Available preview sizes: [4608x3456, 4608x2592, 4608x2304, 4160x3120, 3840x2160, 3840x1920, 3456x3456, 3264x2448, 2592x1940, 2592x1944, 2560x1920, 2304x1728, 2304x1296, 2304x1152, 1840x1380, 1920x1080, 1728x1728, 1440x1080, 1280x960, 1280x720, 1280x480, 1024x768, 720x480, 640x480, 480x360, 480x320, 352x288, 320x240, 176x144]
2020-12-09 17:43:50.876 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: Camera2Session: Available fps ranges: [[10.0:15.0], [10.0:20.0], [7.0:24.0], [24.0:24.0], [7.0:30.0], [30.0:30.0], [10.0:30.0]]
2020-12-09 17:43:50.877 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: Camera2Session: Using capture format: 176x144@[7.0:30.0]
2020-12-09 17:43:50.878 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: Camera2Session: Opening camera 1
2020-12-09 17:43:50.878 21595-21595/com.twilioVideoProto I/AudioManager: In isSpeakerphoneOn(), calling application: com.twilioVideoProto
2020-12-09 17:43:50.878 21595-21595/com.twilioVideoProto I/AudioManager: In setMicrophoneMute(), on: false, calling application: com.twilioVideoProto
2020-12-09 17:43:50.883 21595-26050/com.twilioVideoProto E/libc: Access denied finding property "persist.vendor.camera.privapp.list"
2020-12-09 17:43:50.873 21595-21595/com.twilioVideoProto W/CaptureThread: type=1400 audit(0.0:548302): avc: denied { read } for name="u:object_r:persist_camera_prop:s0" dev="tmpfs" ino=21925 scontext=u:r:untrusted_app_27:s0:c49,c257,c512,c768 tcontext=u:object_r:persist_camera_prop:s0 tclass=file permissive=0
2020-12-09 17:43:50.885 21595-21595/com.twilioVideoProto I/AudioManager: In setMode(), mode: 3, calling application: com.twilioVideoProto
2020-12-09 17:43:50.919 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: Camera2Session: Camera opened.
2020-12-09 17:43:50.931 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: Camera2Session: Camera capture session configured.
2020-12-09 17:43:50.932 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: Camera2Session: Stabilization not available.
2020-12-09 17:43:50.932 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: Camera2Session: Auto-focus is not available.
2020-12-09 17:43:50.933 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: Camera2Session: Camera device successfully started.
2020-12-09 17:43:50.933 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: CameraCapturer: Create session done. Switch state: IDLE
2020-12-09 17:43:50.933 21595-25795/com.twilioVideoProto W/Gralloc3: allocator 3.x is not supported
2020-12-09 17:43:50.934 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: SurfaceTextureHelper: Setting listener to tvi.webrtc.-$$Lambda$Camera2Session$CaptureSessionCallback$avOpmBlgtFuwVDip8hBflrg29nE@f518722
2020-12-09 17:43:50.957 21595-21595/com.twilioVideoProto I/AudioManager: In setSpeakerphoneOn(), on: false, calling application: com.twilioVideoProto
2020-12-09 17:43:51.121 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: SurfaceEglRenderer: : Reporting first rendered frame.
2020-12-09 17:43:51.122 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: SurfaceEglRenderer: : Reporting frame resolution changed to 176x144 with rotation 270
2020-12-09 17:43:51.124 21595-26050/com.twilioVideoProto I/CropperCameraCapturer: 176 144
2020-12-09 17:43:51.124 21595-26050/com.twilioVideoProto I/CropperCameraCapturer: cropX 63 cropY 47 CroppedBuffer Width & Height 50 50
2020-12-09 17:43:52.935 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: CameraStatistics: Camera fps: 1.
2020-12-09 17:43:53.088 21595-25864/com.twilioVideoProto I/ReactNativeJS: CONNECTED
2020-12-09 17:43:54.275 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: InitRecording[tid=26034]
2020-12-09 17:43:54.275 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: ObtainEngineInterface
2020-12-09 17:43:54.276 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: CreateAudioRecorder
2020-12-09 17:43:54.276 21595-26034/com.twilioVideoProto W/libOpenSLES: Conversion from OpenSL ES positional channel mask 0x4 to Android mask 0 loses channels
2020-12-09 17:43:54.276 21595-26034/com.twilioVideoProto I/libOpenSLES: Emulating old channel mask behavior (ignoring positional mask 0x4, using default mask 0x10 based on channel count of 1)
2020-12-09 17:43:54.334 21595-26034/com.twilioVideoProto I/AudioRecord: createRecord_l(125): AUDIO_INPUT_FLAG_FAST successful; frameCount 0 -> 4096
2020-12-09 17:43:54.335 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: StartRecording[tid=26034]
2020-12-09 17:43:54.335 21595-26034/com.twilioVideoProto D/OpenSLESRecorder: state.count:2 state.index:0
2020-12-09 17:43:54.542 21595-26133/com.twilioVideoProto W/OpenSLESRecorder: Bad OpenSL ES record timing, dT=206 [ms]
2020-12-09 17:43:54.770 21595-26043/com.twilioVideoProto I/tvi.webrtc.Logging: EglRenderer: Duration: 4007 ms. Frames received: 1. Dropped: 0. Rendered: 1. Render fps: .2. Average render time: 32577 us. Average swapBuffer time: 3216 us.
2020-12-09 17:43:54.936 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: CameraStatistics: Camera fps: 0.
2020-12-09 17:43:56.936 21595-26050/com.twilioVideoProto I/tvi.webrtc.Logging: CameraStatistics: Camera fps: 0.
2020-12-09 17:43:56.937 21595-26050/com.twilioVideoProto E/tvi.webrtc.Logging: CameraStatistics: Camera freezed.
2020-12-09 17:43:56.946 21595-26050/com.twilioVideoProto E/Camera2Capturer: Camera failure. Client must return video buffers.
    com.twilio.video.Camera2Capturer$Exception: Camera failure. Client must return video buffers.
        at com.twilio.video.Camera2Capturer$1.onCameraFreezed(Camera2Capturer.java:85)
        at tvi.webrtc.CameraVideoCapturer$CameraStatistics$1.run(CameraVideoCapturer.java:131)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at android.os.Looper.loop(Looper.java:214)
        at android.os.HandlerThread.run(HandlerThread.java:67)
2020-12-09 17:43:58.775 21595-26043/com.twilioVideoProto I/tvi.webrtc.Logging: EglRenderer: Duration: 4005 ms. Frames received: 0. Dropped: 0. Rendered: 0. Render fps: .0. Average render time: NA. Average swapBuffer time: NA.
2020-12-09 17:44:02.782 21595-26043/com.twilioVideoProto I/tvi.webrtc.Logging: EglRenderer: Duration: 4007 ms. Frames received: 0. Dropped: 0. Rendered: 0. Render fps: .0. Average render time: NA. Average swapBuffer time: NA.
```
